### PR TITLE
feat: To generate a random string for the JWS id.

### LINKED ISSUE #25: To generate an ID
https://github.com/trustbloc/hub-store/blob/a4b06826e4fc8ecb5e1150e6ba85d2e5cc3d3c66/internal/server/jws.go#L48-L53

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in a4b06826e4fc8ecb5e1150e6ba85d2e5cc3d3c66 when #22 was merged. cc @parasu17.

### LINKED ISSUE #26: Generate ID for the JWS token

### PR DESCRIPTION
**Pull request by parasu17**
Closes #25

Signed-off-by: Parasuraman Ramalingam <Parasuraman.Ramalingam@securekey.com>

### CODE REVIEW

File: internal/crypto/utils.go [hunk 1]
```diff
@@ -20,8 +20,17 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
+	"math/rand"
+	"time"
 )
 
+const (
+	charset  = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
```
Comment (llorllale) [inline internal/crypto/utils.go]: @parasu17 both these constants can be declared inside `RandomString` for better encapsulation

File: internal/crypto/utils.go [hunk 2]
```diff
@@ -20,8 +20,17 @@ import (
...
+	idLength = 20
+)
+
+var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
```
Comment (llorllale) [inline internal/crypto/utils.go]: @parasu17 same here - there's no need for `seededRand` to be global

File: internal/crypto/utils.go [hunk 3]
```diff
@@ -60,3 +62,14 @@ func ParsePrivateKey(data []byte) (interface{}, error) {
 
 	return nil, fmt.Errorf("square/go-jose: parse error: '%s'", err)
 }
+
+// RandomString creates a random string of a constant length
+func RandomString() string {
+	charset := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
```
Comment (sandrask) [inline internal/crypto/utils.go]: use constant for charset

File: internal/crypto/utils.go [hunk 4]
```diff
@@ -60,3 +62,14 @@ func ParsePrivateKey(data []byte) (interface{}, error) {
...
+	seededRand := rand.New(rand.NewSource(time.Now().UnixNano()))
```
Comment (sandrask) [inline internal/crypto/utils.go]: We don't want rand.New() on every call, this is expensive call.

File: internal/crypto/utils.go [hunk 5]
```diff
@@ -60,3 +62,14 @@ func ParsePrivateKey(data []byte) (interface{}, error) {
 
 	return nil, fmt.Errorf("square/go-jose: parse error: '%s'", err)
 }
+
+// RandomString creates a random string of a constant length
+func RandomString() string {
```
Comment (sandrask) [inline internal/crypto/utils.go]: we should not have generating random ID and private/public key utilities in the same file, even better they should be in different package.

File: internal/crypto/utils.go (hunk 3 above)
Comment (parasu17) [reply internal/crypto/utils.go]: That's exactly what I had in my first commit. @llorllale requested to put it locally into the function.

File: internal/crypto/utils.go (hunk 4 above)
Comment (parasu17) [reply internal/crypto/utils.go]: That's exactly what I had in my first commit. @llorllale requested to put it locally into the function.

Comment (sandrask) [reply internal/crypto/utils.go]: You can have different solution where it is not global but instance variable

File: internal/crypto/utils.go (hunk 5 above)
Comment (parasu17) [reply internal/crypto/utils.go]: Reverted back to the previous commit.

File: internal/crypto/utils.go (hunk 4 above)
Comment (parasu17) [reply internal/crypto/utils.go]: Reverted back to the previous commit.

File: internal/crypto/utils.go (hunk 3 above)
Comment (parasu17) [reply internal/crypto/utils.go]: Reverted back to the previous commit.

Comment (llorllale) [reply internal/crypto/utils.go]: @parasu17 constants can be declared inside functions. That was my suggestion.

Comment (parasu17) [reply internal/crypto/utils.go]: Done.

**Pull request merged**